src/Compilers/CSharp/Portable/Emitter/Model/NamedTypeSymbolAdapter.cs

```diff
@@ -668,6 +668,11 @@ IEnumerable<Cci.IMethodDefinition> Cci.ITypeDefinition.GetMethods(EmitContext co
             foreach (var method in this.GetMethodsToEmit())
```

Comment:
>GetMethodsToEmit [](start = 40, length = 16)

I think that the filtering should be done by ```GetMethodsToEmit``` method. The filtering performed by this method is full vs. reference assembly. Other consumers of ```GetMethodsToEmit``` (existing or future) should not see the filtered out constructor.

Reply:
Not yet sure how to implement this part. I am checking if the synthesized static constructor will be emitted using PEModuleBuilder,GetMethodBody. Some of the call sites don't have a PEModuleBuilder around, so some other way to check the constructor is probably necessary.

Maybe I need to add a property on the synthesized constructor symbol which forces binding the static initializers, if it hasn't happened yet?

Reply:
Because the other place where we make this check enumerates all members, not just methods, I ended up doing this by factoring out a ShouldEmit helper for methods.